Client/qtTeamTalk/mainwindow.cpp

```diff
@@ -5093,6 +5096,42 @@ void MainWindow::slotChannelsDeleteFile(bool /*checked =false */)
     }
 }
 
+void MainWindow::slotChannelsShare(bool checked/*=false*/)
+{
+    QClipboard *cp = QApplication::clipboard();
+    QString link = QString("tt://%1?tcpport=%2&udpport=%3&encrypted=%4").arg(m_host.ipaddr).arg(m_host.tcpport).arg(m_host.udpport).arg(m_host.encrypted);
+    bool ok = false;
+    QInputDialog inputDialog;
+    inputDialog.setOkButtonText(tr("&Ok"));
+    inputDialog.setCancelButtonText(tr("&Cancel"));
+    inputDialog.setInputMode(QInputDialog::TextInput);
+    inputDialog.setTextValue(m_host.username);
+    inputDialog.setWindowTitle(tr("Share channel"));
+    inputDialog.setLabelText(tr("Type username to use to share this channel:"));
+    ok = inputDialog.exec();
+    if (inputDialog.textValue().size()>0)
+        link += QString("&username=%5").arg(inputDialog.textValue());

```

Comment:
You need to URL-encode the username.

Reply:
Have you an example for this? I guess it's the QUrl::toPercentEncoding function?